app/models/communitarian/resolution.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative "../../jobs/regular/reopen_resolution"
+require_relative "../post"
+
+module Communitarian
+  class Resolution
+    using Communitarian
+
+    def self.reopened_poll_post_attributes
+      %w[user_id topic_id raw post_type last_editor_id last_version_at user_deleted public_version]
+    end
+
+    def self.reopen_weekly_resolution!(original_post)
+      original_poll = original_post.polls.first
+      return if original_post.user_deleted || !original_post.resolution?
+
+      post_attributes = original_post.attributes.slice(*self.reopened_poll_post_attributes)
+      post = Post.create!(post_attributes)
+      poll = post.reload.polls.first
```

Comment:
It seems like poll will be blank in this case since you created a new post without poll params

Reply:
Yes, but poll params are passed as post body (attribute `raw`).
```ruby
> post_attributes = post_with_poll.attributes.slice(*REOPENED_RESOLUTION_ATTRIBUTES)
=> { "raw" =>
     "[poll type=regular results=always chartType=bar close=2020-08-08T21:00:00.000Z]\n* Yes\n* No\n* Close the poll\n[/poll]",
     ... }
> clone = Post.create!(post_attributes)
=> #<Post:0x00007fc1c2...>
> clone.reload.polls.first
=> #<Poll:0x00007fc1c...>
> clone.reload.polls.first.poll_options
=> [#<PollOption:0x00007fc1c99c3578... html: "Yes",...>,
    #<PollOption:0x00007fc1c99c31e0... html: "No",...>,
    #<PollOption:0x00007fc1c99c2970... html: "Close the poll", ...>]
```